invenio_rdm_records/resources/serializers/marcxml/schema.py

```diff
@@ -0,0 +1,287 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""MARCXML based Schema for Invenio RDM Records."""
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+import traceback
+
+import bleach
+from invenio_access.permissions import system_identity
+from invenio_vocabularies.proxies import current_service as vocabulary_service
+from marshmallow import Schema, fields, missing
+
+from ..ui.schema import current_default_locale
+from ..utils import get_vocabulary_props
+
+
+class MARCXMLSchema(Schema):
+    """Schema for records in MARC."""
+
+    contributors = fields.Method("get_contributors")
+    titles = fields.Method("get_titles")
+    creators = fields.Method("get_creators")
+    identifiers = fields.Method("get_identifiers")
+    relations = fields.Method("get_relations")
+    rights = fields.Method("get_rights")
+    dates = fields.Method("get_dates")
+    subjects = fields.Method("get_subjects")
+    descriptions = fields.Method("get_descriptions")
+    publishers = fields.Method("get_publishers")
+    types = fields.Method("get_types")
+    sources = fields.Method("get_sources")
+    coverage = fields.Method("get_locations")
+    formats = fields.Method("get_formats")
+    parent_id = fields.Method("get_id")  # No GOOD
+    community_id = fields.Method("get_id_community")
+    last_updated = fields.Method("last_updated")
+    sizes = fields.Method("get_sizes")
+    version = fields.Method("get_version")
+    funding = fields.Method("get_funding")
+
+    def get_version(self, obj):
+        """Get the version of the record."""
+        return obj["metadata"].get("version", missing)
+
+    def get_sizes(self, obj):
+        """Get the size of the record."""
+        return obj["metadata"].get("sizes", missing)
+
+    def get_funding(self, obj):
+        """Get funding information."""
+        if "funding" not in obj["metadata"]:
+            return missing
+
+        funders = []
+
+        for funder in obj["metadata"].get("funding", []):
+            funder_string = ""
+
+            scheme = (
```

Comment:
minor extract it outside so you have to calculate it only once, even if access to dict or position in lists is O(1), for readability, would help in many lines below too.

```python
award = funder.get("award", {}).get("identifiers", [{}])[0]

scheme = award.get("scheme", "null")
identifier = award.get("identifier", "null")
```